layouts/community/ergodox/replicaJunction/keymap.c

```diff
@@ -1,427 +1,381 @@
+//                                                     _______,
+//                                     _______,_______,_______,
+
+//     _______,_______,_______,_______,_______,_______,_______,
+//     _______,_______,_______,_______,_______,_______,_______,
+//             _______,_______,_______,_______,_______,_______,
+//     _______,_______,_______,_______,_______,_______,_______,
+//                     _______,_______,_______,_______,_______,
+//     _______,_______,
+//     _______,
+//     _______,_______,_______
+// ),
 
 };
 
-// The normal QMK functions ending in _user are overridden in the
-// replicaJunction userspace. Those functions handle my global
-// settings, and redirect to these _keymap functions if something
-// is unhandled. This allows me to keep most of my global preferences
-// in one place while still allowing keyboard-specific code.
+void keyboard_post_init_user_kb(void) {
+    ergodox_right_led_1_on();
+}
+
+
+////////////////////////////////////////////////////////////
+// System LED code
+////////////////////////////////////////////////////////////
+
+// We need to track both the layer state and the system LED state in
+// order to update the LEDs appropriately when either one changes.
+// These are both represented as a bitmask.
+//
+// There is a global 'layer_state' variable but it is set after the call
+// to layer_state_set_user(), so it doesn't work properly for our
+// purposes.
+static uint32_t current_layer_state = 0;
+static uint8_t sys_led_state = 0;
+
+// Use these masks to read the system LEDs state.
+static const uint8_t sys_led_mask_num_lock = 1 << USB_LED_NUM_LOCK;
+static const uint8_t sys_led_mask_caps_lock = 1 << USB_LED_CAPS_LOCK;
+static const uint8_t sys_led_mask_scroll_lock = 1 << USB_LED_SCROLL_LOCK;
+
+// Whether the given layer (one of the constant defined at the top) is active.
+#define LAYER_ON(state, layer) (state & (1<<layer))
+
+// Brightness of LEDs (0-255)
+// Default value is 255. Use a lesser value for dimmer LEDs.
+static const uint8_t max_led_value = 255;
+
+void led_1_on(void) {
+    ergodox_right_led_1_on();
+    ergodox_right_led_1_set(max_led_value);
+}
+
+void led_2_on(void) {
+    ergodox_right_led_2_on();
+    ergodox_right_led_2_set(max_led_value);
+}
+
+void led_3_on(void) {
+    ergodox_right_led_3_on();
+    ergodox_right_led_3_set(max_led_value);
+}
+
+// The onboard LED doesn't support brightness, but this function keeps
+// things consistent.
+void led_board_on(void) {
+    ergodox_board_led_on();
+}
+
+void led_1_off(void) {
+    ergodox_right_led_1_off();
+}
+
+void led_2_off(void) {
+    ergodox_right_led_2_off();
+}
+
+void led_3_off(void) {
+    ergodox_right_led_3_off();
+}
+
+void led_board_off(void) {
+    ergodox_board_led_off();
+}
 
-// The idea was shamelessly copied from the amazing Drashna.
+void set_leds_by_system_state(uint8_t led_state) {
+    if (led_state & sys_led_mask_num_lock) {
```

Comment:
You should use `host_keyboard_led_state().num_lock` etc. here instead.

Reply:
Oh neat, I didn't know about that. Thanks!

Changed in 3d306fb